images/airflow/2.9.2/python/mwaa/subprocess/subprocess.py

```diff
@@ -32,6 +33,10 @@
 # from us before we forcefully terminate the process with a SIGKILL.
 _SIGTERM_DEFAULT_PATIENCE_INTERVAL = timedelta(seconds=90)
 
+# The time for which the log reader will sleep when subprocess is still running
+# But last log read was empty.
+_SUBPROCESS_LOG_POLL_IDLE_SLEEP_INTERVAL = timedelta(milliseconds=50)
```

Comment:
Is it too robust given that before we sleep roughly 1 second between each loop?

Reply:
Yeah, this is something to considered. The issue here is that even a small environment can be a noisy logger. The goal here would be to poll faster than logs are produced by airflow. Which means we are trying to track an unknown quantity, since this will vary component to component, environment to environment, and also vary over time for same component / environment. 

Since the sleep is only when buffer is empty, maybe an increase to 100ms is warranted to be more cautious.